python/lsst/daf/butler/tests/_testRepo.py

```diff
@@ -107,7 +107,8 @@ def makeTestRepo(
     butler = Butler(newConfig, writeable=True)
     dimensionRecords = _makeRecords(dataIds, butler.registry.dimensions)
     for dimension, records in dimensionRecords.items():
-        butler.registry.insertDimensionData(dimension, *records)
+        if butler.registry.dimensions[dimension].viewOf is None:
+            butler.registry.insertDimensionData(dimension, *records)
```

Comment:
It may be that it was already possible via some other method, but I made this change because passing band key in the `dataIds` argument did result in an attempt to insert it, which failed.  I suspect that I missed a way to make it unnecessary to pass a band key here, even after the NOT NULL constraint lands - and I'll discuss that in a thread for another comment - but even if that's the case it seems like it might still be good to permit a `band` key to be passed here, and allow its values to be associated with other dimensions like `physical_filter`, in the same way other relationships are generated arbitrarily.